Sources/Brave/States/AppState.swift

```diff
@@ -0,0 +1,221 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+import BraveCore
+import Data
+import RuntimeWarnings
+import Shared
+import Growth
+import Preferences
+import Storage
+import os.log
+
+/// Class that does startup initialization
+/// Everything in this class can only be execute ONCE
+/// IE: BraveCore initialization, BuildChannel, Migrations, etc.
+public class AppState {
+  public static let shared = AppState()
```

Comment:
Lets move this to the actual application Xcode target (`App/iOS` folder with the AppDelegate/SceneDelegate). I want to avoid this existing outside of AppDelegate/SceneDelegate otherwise someone is going to reach into this at some point and abuse it